### PR TITLE
Mobile kiosk mode

### LINKED ISSUE #812: Add the ability to use kiosk mode form a mobile phone, use camera to capture barcode on IDs

### PR DESCRIPTION
**Pull request by Mercy-Eze**
Barcode function works (literally)... but not technically
Outputs False Positives

### CODE REVIEW

**Comment by AndersonStettner** [comment]
[Add the ability to use kiosk mode form a mobile phone, use camera to capture barcode on IDs #812](https://trello.com/c/99d5seQZ/583-add-the-ability-to-use-kiosk-mode-form-a-mobile-phone-use-camera-to-capture-barcode-on-ids-812)

File: app/templates/base.html [hunk 1]
```diff
@@ -101,6 +101,7 @@
 {% block scripts %}
   <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
   <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/quagga/dist/quagga.min.js"></script>
```
Comment (BrianRamsay) [inline app/templates/base.html]: we aren't using quagga anymore

File: app/static/js/eventKiosk.js [hunk 2]
```diff
@@ -51,6 +51,43 @@ function eventFlasher(flash_message, status){
...
+  }
+  var html5QrcodeScanner = new Html5QrcodeScanner(
+      "qr-reader", { fps: 15, qrbox : qrboxFunction, preferFrontCamera: false });
+  html5QrcodeScanner.render(onScanSuccess);
+
+  $('#html5-qrcode-button-camera-stop').on("click", function() {
+    $('#qr-reader').hide()
+    })
+  })
+
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: Let's leave some comments in this section that explain what the pieces do

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
1. You still have to hit the camera button twice to bring up the camera
2. Let's make the camera button a little larger (side padding)
3. The placeholder text is cut off on my phone "Enter scanner da". I'd change it to something like Enter B-Number or B# anyway.
4. Using the full screen option hides the camera button
5. Sometimes the camera button is the same width as the input box and below it, and sometimes it is small on the right. We should be consistent.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
Make sure to handle the same person being added multiple times.

File: app/static/js/eventKiosk.js [hunk 3]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
...
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  var lastResult, countResults = 0;
+  function onScanSuccess(decodedText, decodedResult) {
+      if (decodedText !== lastResult) {
+          ++countResults;
+          lastResult = decodedText;
+          // Handle on success condition with the decoded message.
+          console.log(`Scan result ${decodedText}`, decodedResult);
```
Comment (AndersonStettner) [inline app/static/js/eventKiosk.js]: Remove debug output

File: app/static/js/eventKiosk.js [hunk 4]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
 
 }
 
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  var lastResult, countResults = 0;
```
Comment (zachneill) [inline app/static/js/eventKiosk.js]: hey dont use var again

File: app/static/js/eventKiosk.js [hunk 5]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
...
+  let qrboxFunction = function(viewfinderWidth, viewfinderHeight) {
+    let minEdgePercentage = 0.9; // 90%
+    let minEdgeSize = Math.min(viewfinderWidth, viewfinderHeight);
+    let qrboxSize = Math.floor(minEdgeSize * minEdgePercentage);
+    return {
+        width: qrboxSize,
+        height: qrboxSize
+    };
+  }
+  var html5QrcodeScanner = new Html5QrcodeScanner(
```
Comment (zachneill) [inline app/static/js/eventKiosk.js]: gah you used var again

**Review by zachneill — CHANGES_REQUESTED** [review]
var is gonna be deprecated one day

File: app/static/js/eventKiosk.js [hunk 6]
```diff
@@ -4,7 +4,9 @@ $(document).keydown(function(e){
     toggleFullscreen();
   }
   else if(e.key === "Enter") {
-      submitData(true);
+      submitData(true)
+      $('.qr-reader-button').function(true)
+      ;
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: What does this do?

File: app/static/js/eventKiosk.js [hunk 7]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
 
 }
 
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  let lastResult, countResults = 0;
+  function onScanSuccess(decodedText, decodedResult) {
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: I would add a check to make sure we have a full b-number, to avoid submitting that case where it scanned a partial one

File: app/static/js/eventKiosk.js [hunk 8]
```diff
@@ -61,18 +97,22 @@ function submitData(hitEnter = false){
         "eventid": $("#eventid").val(),
         "bNumber": $("#submitScannerData").val()
       },
-      success: function(result) {
+      success: function(decodedText) {
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: what is passed into this function is not decoded text. If you are going to rename it from result, name it something more meaningful.

File: app/static/js/eventKiosk.js [hunk 9]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
...
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  let lastResult, countResults = 0;
+  function onScanSuccess(decodedText, decodedResult) {
+      if (decodedText !== lastResult) {
+          ++countResults;
+          lastResult = decodedText;
+          // Handle on success condition with the decoded message.
+          console.log(`Scan result ${decodedText}`, decodedResult);
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: Remove debug output

File: app/static/js/eventKiosk.js [hunk 10]
```diff
@@ -51,6 +53,40 @@ function eventFlasher(flash_message, status){
 
 }
 
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  let lastResult, countResults = 0;
+  function onScanSuccess(decodedText, decodedResult) {
+      if (decodedText !== lastResult) {
+          ++countResults;
```
Comment (BrianRamsay) [inline app/static/js/eventKiosk.js]: You don't use `countResults`

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

File: app/static/js/eventKiosk.js [hunk 11]
```diff
@@ -51,6 +52,42 @@ function eventFlasher(flash_message, status){
 
 }
 
+$('.qr-reader-button').on("click", function() {
+  //  Opens the camera to scan the ID
+  $('#qr-reader').toggle()
+  let lastResult, countResults = 0;
+  function onScanSuccess(decodedText, decodedResult) {
+    console.log('id code', decodedText)
```
Comment (AndersonStettner) [inline app/static/js/eventKiosk.js]: Remove debug output

File: app/static/js/eventKiosk.js [hunk 12]
```diff
@@ -51,6 +52,42 @@ function eventFlasher(flash_message, status){
...
+      if (decodedText && decodedText.length > 9 && decodedText !== lastResult) {
+          lastResult = decodedText;
+        
+          
+          $("#submitScannerData").val(decodedText)
+          submitData(true);
+      } else {
+          // Handle the case where the scanned data is not a full B-number.
+        console.log("Invalid B-number:", decodedText);
```
Comment (AndersonStettner) [inline app/static/js/eventKiosk.js]: Remove debug output

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

**Review by BrianRamsay — APPROVED** [review]
